### DISCUSSION TITLE
Testing this from external

### DISCUSSION THREAD

**Discussion by theCJMan** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#76] [Q&A] [answered]
Hi.
How do you propose i test this app inc webhooks while it runs on my local pc (dev environment).
I'm using win 11.

I tried NGrok, but the app is not loading externally.

Thanx for your feedback

**Comment by gabrieldwight** [comment]
Hi, you can use the conveyor in Visual Studio by installing the extension or dev tunnels built in Visual Studio 2022.

**Reply by theCJMan** [reply]
Thank you, I got NGrok to work... But i will check out your solution as well

I had to adapt the command as below for NGrok
ngrok http https://localhost:<port for https as per our project>

e.g.
ngrok http https://localhost:7723

**Comment by theCJMan** [comment, marked as answer]
Hey All
I am back after some months and need to run my code again, I reconfig my FB WebHook, and found a few issues...
Here are a few pointers:
Using DevTunnels within Visual Studio:
1. Make sure your Tunnel is Public - it seems a private tunnel is not working well.
2. Make the Tunnel Persisten, else you will need to reconfig your tunnel every time at FB

Using NGrok as stated above:
1. Use: ngrok http https://localhost:7723/
2. Make sure the port is correct
